chapter_5/exercise_5_06/getline.c

```diff
@@ -1,23 +1,19 @@
 #include <stdio.h>
 
-void get_line(char *s);
 
-int main(void)
-{
-  char string[150] = "";
+char *getln(char *line);
 
-  get_line(string);
-  puts(string);
-
-  return 0;
+int main() {
+    char line[100];
+    
+    getln(line);
+    printf("%s", line);
+    
+    return 0;
 }
 
-void get_line(char *s)
-{
-  while ((*s = getchar()) != EOF && (*s != '\n'))
-  {
-    ++s;
-  }
-
-  *s = '\0';
+char *getln(char *line) {
+    while ((*line++ = getchar()) != 10);
```

Comment:
@kyopa I see that your approach is shorter, which is very nice. However, you'll read up to the end of the file, while the idea was to read a line, even though there might be just only a short line that's the entire file. So, in fact, the original solution is better than your shorter proposal.

If you could make one line of code that preserves the original functionality it'll be great.